Switch containers/[image,storage,common] to go.podman.io.

jankaluza opened the pull request:
These go packages were migrated to a monorepo, as stated in the https://blog.podman.io/2025/08/migration-to-the-container-libs-monorepo-is-complete/.

This commit updates the crc package to use these packages from new locations.

It was generated using following commands:

```
$ find . -type f -name '*.go' -exec sed -i -e 's,"github.com/containers/image,"go.podman.io/image,g' {} \;
$ find . -type f -name '*.go' -exec sed -i -e 's,"github.com/containers/common,"go.podman.io/common,g' {} \;
$ find . -type f -name '*.go' -exec sed -i -e 's,"github.com/containers/storage,"go.podman.io/storage,g' {} \;
$ goimports -v -w .
$ git checkout vendor/
$ git checkout tools/vendor
$ vi go.mod # to change the storage, image and common imports
$ go mod tidy
$ go mod vendor
```

## Description

See the description above.

## Type of change
<!--
What types of changes does your code introduce? Put an `x` in all the boxes that apply
-->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change
- [x] Chore (non-breaking change which doesn't affect codebase;
  test, version modification, documentation, etc.)

## Proposed changes
<!--
List main as well as consequential changes you introduced or had to introduce.

1. Add `start` command.
2. Add `setup` as prerequisite to `start`.
3. ...
-->

## Testing
<!--
What is the _bottom-line_ functionality that needs testing? Describe in pseudo-code or in English. Use verifiable statements that tie your changes to existing functionality.

1. `start` succeeds first time after `setup` succeeded
2. stdout contains ... if `start` succeeded
3. stderr contains ... after `start` failed
4. `status` returns ... if `start` succeeded
5. `status` returns ... if `start` failed
6. `start` fails after `start` succeeded or after `status` says CRC is `Running`
7. ...
-->

## Contribution Checklist
- [x] I Keep It Small and Simple: The smaller the PR is, the easier it is to review and have it merged
- [x] I have performed a self-review of my code
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Which platform have you tested the code changes on? <!-- Only put an `x` in applicable platforms -->
    - [x] Linux
    - [ ] Windows
    - [ ] MacOS

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- Security
  - Strengthened TLS defaults; CBC cipher suites removed.
  - Support for passphrase-encrypted TLS private keys deprecated and no longer accepted.
- Chores
  - Migrated container/image and strongunits dependencies to go.podman.io modules.
  - Updated multiple third-party libraries; improved Docker client TLS configuration defaults.
  - Bumped Go to 1.24.2 (toolchain 1.24.6).
- Deprecations
  - SQLite user authentication is deprecated in the bundled driver.
- Compatibility
  - Image handling now uses Podman-maintained modules; no expected behavior changes.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

jankaluza commented:
Note that `github.com/containers/common v0.62.2 // indirect` is still there. This is pulled in by the vfkit. I'm preparing PR also for vfkit, but since there is a circular dependency between vfkit and crc, we will need to merge both PRs first before the containers/common dependency disappears forever.

redbeam commented:
/ok-to-test

redbeam commented:
Hi @jankaluza, thanks for this PR. In the meantime, some conflicts appeared, can you rebase your branch?

jankaluza committed:
Switch containers/[image,storage,common] to go.podman.io

These go packages were migrated to a monorepo, as stated in the
https://blog.podman.io/2025/08/migration-to-the-container-libs-monorepo-is-complete/.

This commit updates the crc package to use these packages from new locations.

It was generated using following commands:

```
$ find . -type f -name '*.go' -exec sed -i -e 's,"github.com/containers/image,"go.podman.io/image,g' {} \;
$ find . -type f -name '*.go' -exec sed -i -e 's,"github.com/containers/common,"go.podman.io/common,g' {} \;
$ find . -type f -name '*.go' -exec sed -i -e 's,"github.com/containers/storage,"go.podman.io/storage,g' {} \;
$ goimports -v -w .
$ git checkout vendor/
$ git checkout tools/vendor
$ vi go.mod # to change the storage, image and common imports
$ go mod tidy
$ go mod vendor
```

Signed-off-by: Jan Kaluza <jkaluza@redhat.com>

jankaluza commented:
@redbeam , rebased.

redbeam commented:
/retest

redbeam commented:
@jankaluza Closing this in favor of #4954 (reason is in that PR).
